### DISCUSSION TITLE
Dependencies added in 0.3.0

### DISCUSSION THREAD

**Discussion by nicobatalla** [jotaijs/jotai-devtools#40] [Q&A]
It looks like 0.3.0 adds a bunch of dependencies that weren't there before

This is a screenshot from the update PR in our app:
![image](https://user-images.githubusercontent.com/25337921/225127540-c42d5f07-c5ee-47a6-9315-d1d7576aa690.png)

Dependencies are colliding with our own styling solution which is `styled-components`, causing some TS errors that weren't there before (I suspect this is coming from the added `@emotion` stuff that gets installed, or from `@mantine`)

Are all the dependencies added really needed?

**Comment by arjunvegda** [comment]
> Dependencies are colliding with our own styling solution which is styled-components, causing some TS errors that weren't there before

Interesting! I'm unaware of interop issues with `styled-components` and `@emotion`. So I'm curious — what kind of errors are you running into? Can you provide code snippets, screenshots, or a small repro? 

You also bring up a valid point about `@emotion`, I think we should move `@emotion/react` as a peer-dependency rather than a dependency 🤔 .

May I ask you to give this a shot? Ref - #42 

```sh
yarn add jotai-devtools@https://pkg.csb.dev/jotaijs/jotai-devtools/commit/58dc03d4/jotai-devtools/_pkg.tgz

# optionally add
yarn add @emotion/react
```

> Are all the dependencies added really needed?

Yes, they power our UI DevTools. Rest assured, `<DevTools/>` is optimized to _not_ get included in your production build. It should be tree-shaken by your bundler.

**Comment by nicobatalla** [comment]
Actually this was an issue on our side.

We are using [MUI](https://mui.com/material-ui/getting-started/installation/) which uses emotion by default, but it can be configured also with `styled-components` as we are doing. That's all good but in one file we were importing the `css` function from `mui` instead of `styled-components` as we do in all others. 

Adding `@emotion/react` made this apparent because now the `css` function was taking the types from `emotion` which are different that the ones from `styled-components`. So we just fixed the import in that file to use `css` from `styled-components` as it should and no types issues were found.

Also, I checked our production build size just in case with 0.3.0 and it is the same as before, so we are ok on that front too.

**Reply by arjunvegda** [reply]
Great! Glad to hear that it is sorted 🙌
